# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1384630714 -3600
#      Sat Nov 16 20:38:34 2013 +0100
# Node ID 9799a996b2c1a7459ddb88adfc0c994a81022b32
# Parent  b3e4ee8f4d6d0e0b9e649c7fc74fd04d4b69e225
* file-editor.cc (request_open_file): new edit tab when file name is empty

diff --git a/libgui/src/m-editor/file-editor.cc b/libgui/src/m-editor/file-editor.cc
--- a/libgui/src/m-editor/file-editor.cc
+++ b/libgui/src/m-editor/file-editor.cc
@@ -269,18 +269,19 @@ file_editor::request_open_file (const QS
                                 bool debug_pointer,
                                 bool breakpoint_marker, bool insert)
 {
   if (call_custom_editor (openFileName, line))
     return;   // custom editor called
 
   if (openFileName.isEmpty ())
     {
-      // ??  Not sure this will happen.  This routine isn't even called
-      // if the user hasn't selected a file.
+      // This happens if edit is calles without an argument
+      // Open eitor with empty edit area instead (as new file would do)
+      request_new_file ("");
     }
   else
     {
       // Have all file editor tabs signal what their file names are.
       editor_tab_map.clear ();
       emit fetab_file_name_query (0);
 
       // Check whether this file is already open in the editor.
